crates/bevy_animation_rig/src/skinned_mesh.rs

```diff
@@ -0,0 +1,215 @@
+use bevy_render::{
+    pipeline::PipelineDescriptor,
+    render_graph::{base::node, RenderGraph, RenderResourcesNode},
+    renderer::{
+        RenderResource, RenderResourceHints, RenderResourceIterator, RenderResourceType,
+        RenderResources,
+    },
+    shader::{Shader, ShaderStage},
+    texture::Texture,
+};
+use bevy_transform::components::GlobalTransform;
+
+/// Specify RenderPipelines with this handle to render the skinned mesh.
+pub const SKINNED_MESH_PIPELINE_HANDLE: HandleUntyped =
+    HandleUntyped::weak_from_u64(PipelineDescriptor::TYPE_UUID, 0x14db1922328e7fcc);
+
+/// Used to update and bind joint transforms to the skinned mesh render pipeline specified with [`SKINNED_MESH_PIPELINE_HANDLE`].
+///
+/// The length of `joint_entities` and `joint_transforms` should equal to the number of matrices inside [`SkinnedMeshInverseBindposes`].
+///
+/// The content of `joint_transforms` can be modified manually if [`skinned_mesh_update`] system is disabled.
+///
+/// # Example
+/// ```
+/// use bevy_animation_rig::{SkinnedMesh, SKINNED_MESH_PIPELINE_HANDLE};
+/// use bevy_ecs::{entity::Entity, system::Commands};
+/// use bevy_pbr::prelude::PbrBundle;
+/// use bevy_render::pipeline::{RenderPipeline, RenderPipelines};
+///
+/// fn example_system(mut commands: Commands) {
+///     commands.spawn_bundle(PbrBundle {
+///         render_pipelines: RenderPipelines::from_pipelines(
+///             vec![RenderPipeline::new(SKINNED_MESH_PIPELINE_HANDLE.typed())]
+///         ),
+///         ..Default::default()
+///     }).insert(SkinnedMesh::new(
+///         // Refer to [`SkinnedMeshInverseBindposes`] example on how to create inverse bindposes data.
+///         Default::default(),
+///         // Specify joint entities here.
+///         vec![Entity::new(0)]
+///     ));
+/// }
+/// ```
+#[derive(Debug, Default, Clone, Reflect)]
+#[reflect(Component, MapEntities)]
+pub struct SkinnedMesh {
+    pub inverse_bindposes: Handle<SkinnedMeshInverseBindposes>,
+    pub joint_entities: Vec<Entity>,
+    pub joint_transforms: Vec<Mat4>,
+}
+
+impl SkinnedMesh {
+    pub fn new(
+        inverse_bindposes: Handle<SkinnedMeshInverseBindposes>,
+        joint_entities: Vec<Entity>,
+    ) -> Self {
+        let entities_count = joint_entities.len();
+
+        Self {
+            inverse_bindposes,
+            joint_entities,
+            joint_transforms: vec![Mat4::IDENTITY; entities_count],
+        }
+    }
+}
+
+impl MapEntities for SkinnedMesh {
+    fn map_entities(&mut self, entity_map: &EntityMap) -> Result<(), MapEntitiesError> {
+        for entity in &mut self.joint_entities {
+            *entity = entity_map.get(*entity)?;
+        }
+
+        Ok(())
+    }
+}
+
+impl RenderResource for SkinnedMesh {
+    fn resource_type(&self) -> Option<RenderResourceType> {
+        Some(RenderResourceType::Buffer)
+    }
+
+    fn write_buffer_bytes(&self, buffer: &mut [u8]) {
+        let transform_size = std::mem::size_of::<[f32; 16]>();
```

Comment:
If we're checking for the transform size, why are use using `[f32; 16]` instead of `Mat4`? 
(I might not be understanding something so please correct me if I'm wrong haha :)

Reply:
I got this `std::mem::size_of::<[f32; 16]>()` from `RenderResource` implementation for `Mat4`. I'm not sure why as well. Maybe `Mat4` may have internal states so we don't rely on the size of `Mat4`?